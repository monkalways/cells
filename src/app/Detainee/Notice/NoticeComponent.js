import React from 'react';
import PropTypes from 'prop-types';
import {
  Avatar, Grid, Typography, withStyles,
} from '@material-ui/core';

import CautionPresentIcon from '../../../images/CautionPresent.png';
import ContagiousIcon from '../../../images/Contagious.png';
import KeepAloneIcon from '../../../images/KeepAlone.png';
import MedicineAcceptIcon from '../../../images/MedicineAccept.png';
import OtherWarningIcon from '../../../images/OtherWarning.png';
import SuicideWarningIcon from '../../../images/SuicideWarning.png';

const CAUTION_PRESENT = 'CautionPresent';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  detainee: PropTypes.shape({
    withCaution: PropTypes.bool.isRequired,
    mustBeKeptAlone: PropTypes.bool.isRequired,
    isSuicidal: PropTypes.bool.isRequired,
    isContagious: PropTypes.bool.isRequired,
    hasWarning: PropTypes.bool.isRequired,
    isUnderMedication: PropTypes.bool.isRequired,
    cautionsArray: PropTypes.arrayOf(PropTypes.string).isRequired,
  }).isRequired,
};

const NoticeComponent = ({ classes, detainee }) => {
  const renderWarnings = () => {
    const icons = [];
    if (detainee.withCaution) icons.push(CautionPresentIcon);
    if (detainee.mustBeKeptAlone) icons.push(KeepAloneIcon);
    if (detainee.isSuicidal) icons.push(SuicideWarningIcon);
    if (detainee.isContagious) icons.push(ContagiousIcon);
    if (detainee.hasWarning) icons.push(OtherWarningIcon);
    if (detainee.isUnderMedication) icons.push(MedicineAcceptIcon);

    return icons.map((icon) => {
      if (icon.includes(CAUTION_PRESENT)) {
        return (
          <Grid item key={icon} xs={1} className={classes.gridItem}>
            <Avatar className={classes.avatar} src={icon} />
            <Typography align="center" className={classes.cautionText}>
              {detainee.cautionsArray.join('')}
            </Typography>
          </Grid>
        );
      }
      return (
        <Grid item key={icon} xs={1} className={classes.gridItem}>
          <Avatar className={classes.avatar} src={icon} />
        </Grid>
      );
    });
  };

  return (
    <Grid container className={classes.grid}>
      <Grid item xs={12}>
        <Typography variant="h6" className={classes.text}>
          Notice
        </Typography>
      </Grid>
      <Grid container item xs={12}>
        {renderWarnings()}
      </Grid>
    </Grid>
  );
};

NoticeComponent.propTypes = propTypes;

export default withStyles((theme) => ({
  avatar: {
    height: theme.spacing.unit * 6,
    width: theme.spacing.unit * 6,
    zIndex: 99,
  },
  cautionText: {
    color: '#FFFFFF',
    backgroundColor: '#F80205',
    width: theme.spacing.unit * 6,
    fontWeight: theme.typography.caption.fontWeight,
    fontSize: theme.typography.caption.fontSize,
  },
  grid: {
    padding: theme.spacing.unit * 0.5,
    minHeight: theme.spacing.unit * 12,
  },
  gridItem: {
    padding: theme.spacing.unit * 0.5,
  },
  text: {
    textDecoration: 'underline',
  },
}))(NoticeComponent);
