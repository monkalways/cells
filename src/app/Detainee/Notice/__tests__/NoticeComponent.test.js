import React from 'react';
import { shallow } from 'enzyme';

import { Avatar, Typography } from '@material-ui/core';
import { NoticeComponent } from '../NoticeComponent';

describe('NoticeComponent', () => {
  let detainee;

  beforeEach(() => {
    detainee = {
      cautionsArray: [],
      hasWarning: false,
      isContagious: false,
      isSuicidal: false,
      isUnderMedication: false,
      mustBeKeptAlone: false,
      withCaution: false,
    };
  });

  const setup = () => {
    const classes = {
      avatar: 'avatar',
      cautionText: 'cautionText',
      grid: 'grid',
      gridItem: 'gridItem',
      text: 'text',
    };
    return shallow(<NoticeComponent classes={classes} detainee={detainee} />);
  };

  it('should render', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  describe('renderWarnings', () => {
    it('should render keep alone warning icon', () => {
      detainee.mustBeKeptAlone = true;
      const wrapper = setup();
      expect(wrapper.find(Avatar)).toHaveProp('src', 'KeepAlone.png');
    });

    it('should render suicide warning icon', () => {
      detainee.isSuicidal = true;
      const wrapper = setup();
      expect(wrapper.find(Avatar)).toHaveProp('src', 'SuicideWarning.png');
    });

    it('should render contagious warning icon', () => {
      detainee.isContagious = true;
      const wrapper = setup();
      expect(wrapper.find(Avatar)).toHaveProp('src', 'Contagious.png');
    });

    it('should render medication warning icon', () => {
      detainee.isUnderMedication = true;
      const wrapper = setup();
      expect(wrapper.find(Avatar)).toHaveProp('src', 'MedicineAccept.png');
    });

    it('should render other warning icon', () => {
      detainee.hasWarning = true;
      const wrapper = setup();
      expect(wrapper.find(Avatar)).toHaveProp('src', 'OtherWarning.png');
    });

    it('should render caution present icon and caution array', () => {
      detainee.withCaution = true;
      detainee.cautionsArray = ['V', 'M', 'E', 'D', 'A'];
      const wrapper = setup();
      expect(wrapper.find(Avatar)).toHaveProp('src', 'CautionPresent.png');
      expect(wrapper
        .find(Typography)
        .at(1)
        .html()).toContain('ADEMV');
    });
  });
});
