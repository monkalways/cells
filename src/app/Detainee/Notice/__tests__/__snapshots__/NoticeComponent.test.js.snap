// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoticeComponent should render 1`] = `
<WithStyles(Grid)
  className="grid"
  container={true}
>
  <WithStyles(Grid)
    item={true}
    xs={12}
  >
    <WithStyles(Typography)
      className="text"
      variant="h5"
    >
      Notice
    </WithStyles(Typography)>
  </WithStyles(Grid)>
  <WithStyles(Grid)
    container={true}
    item={true}
    xs={12}
  />
</WithStyles(Grid)>
`;
